### PR TITLE
chore: stop using `phpunitgoodpractices/traits`

### PR DESCRIPTION
**Pull request by keradus**
ref #6884
built on top of #7362

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -15,12 +15,6 @@
 namespace PhpCsFixer\Tests;
 
 use PHPUnit\Framework\TestCase as BaseTestCase;
-use PHPUnitGoodPractices\Polyfill\PolyfillTrait;
-use PHPUnitGoodPractices\Traits\ExpectationViaCodeOverAnnotationTrait;
```
Comment (keradus) [inline tests/TestCase.php:19]: replaced with already used `php_unit_expectation` rule

File: tests/TestCase.php [hunk 2]
```diff
@@ -15,12 +15,6 @@
...
-use PHPUnitGoodPractices\Traits\ExpectOverSetExceptionTrait;
```
Comment (keradus) [inline tests/TestCase.php:20]: replaced with already used `php_unit_expectation` rule

File: tests/TestCase.php (hunk 1 above)
Comment (keradus) [inline tests/TestCase.php:19]: replaced with already used `php_unit_no_expectation_annotation` rule

File: tests/TestCase.php [hunk 3]
```diff
@@ -15,12 +15,6 @@
...
-use PHPUnitGoodPractices\Traits\IdentityOverEqualityTrait;
```
Comment (keradus) [inline tests/TestCase.php:21]: phpstan-phpunit should already cover, if I recall right

File: tests/TestCase.php [hunk 4]
```diff
@@ -15,12 +15,6 @@
...
-use PHPUnitGoodPractices\Traits\ProphecyOverMockObjectTrait;
-use PHPUnitGoodPractices\Traits\ProphesizeOnlyInterfaceTrait;
```
Comment (keradus) [inline tests/TestCase.php:23]: not sure if anything covers this?

**Review by keradus — COMMENTED** [review]
.

File: tests/TestCase.php [hunk 5]
```diff
@@ -15,12 +15,6 @@
 namespace PhpCsFixer\Tests;
 
 use PHPUnit\Framework\TestCase as BaseTestCase;
-use PHPUnitGoodPractices\Polyfill\PolyfillTrait;
-use PHPUnitGoodPractices\Traits\ExpectationViaCodeOverAnnotationTrait;
-use PHPUnitGoodPractices\Traits\ExpectOverSetExceptionTrait;
-use PHPUnitGoodPractices\Traits\IdentityOverEqualityTrait;
-use PHPUnitGoodPractices\Traits\ProphecyOverMockObjectTrait;
```
Comment (keradus) [inline tests/TestCase.php:22]: replaced by `testNoPHPUnitMockUsed` test

**Review by Wirone — APPROVED** [review]
YEAH!! 🎉

**Pull request merged**